lib/ansible/modules/system/known_hosts.py

```diff
@@ -240,7 +243,7 @@ def search_for_host_key(module,host,key,hash_host,path,sshkeygen):
 
     sshkeygen_command.insert(1,'-H')
     rc,stdout,stderr=module.run_command(sshkeygen_command,check_rc=False)
-    if rc!=0: #something went wrong
+    if rc not in (0, 1) or stderr != '': #something went wrong
```

Comment:
Yes, this is a bug in ssh-keygen.  The commit message of 5eef669313594c809dfe66f27786618409ed91a0 has all the gory details.